crates/turborepo-lib/src/graph/walker.rs

```diff
@@ -282,4 +282,68 @@ mod test {
         walker.wait().await.unwrap();
         assert_eq!(visited.lock().unwrap().as_slice(), &[c, b, e, d, a]);
     }
+
+    // test that long running nodes block dependents, but others can continue
+    #[tokio::test]
+    async fn test_multiple_roots() {
+        // a -- b -- c
+        //   \
+        // d - e
```

Comment:
I'm mildly concerned that the merging branch is from root, but I'm ~sure it works anyway. This is the "slightly harder" variant.

        // a -- b -- c
        //          /
        // d -- e -